rmf_fleet_adapter/src/rmf_fleet_adapter/agv/EasyFullControl.cpp

```diff
@@ -1055,9 +1055,21 @@ void EasyCommandHandle::follow_new_path(
   for (std::size_t i = 0; i < waypoints.size(); ++i)
   {
     const auto& wp = waypoints[i];
-    if (wp.graph_index().has_value())
+    for (const auto& l : current_location)
     {
-      for (const auto& l : current_location)
+      if (!nav_params->skip_rotation_commands)
+      {
+        // If the waypoint and robot orientations are not aligned with
+        // skip rotation command set to False, we will not consider this as
+        // a connection
+        if (std::abs(wp.position()[2] -
+          l.orientation())*180.0 / M_PI > 1e-2)
+        {
+          continue;
```

Comment:
Note that because there are two nested for loops this `continue` will affect the inner one, which is the for loop for `current_location` and not the one that is iterating `waypoints`, is that what you meant to do? The same applies to the other snippet

Reply:
Yep! We'd still want to check this waypoint against other possible current locations (e.g. on lane A, or lane B, or another RMF waypoint) of the robot. While it is likely consistent, I'm not sure the locations in `current_location` would _always_ have the same orientation, so iterating through all combis to be safe.